NumPy 1.23 dropped numpy.asscalar()

## 🐛 Bug

numpy has depricated asscalar() functionality

## To Reproduce

Steps to reproduce the behavior:

1. install dgl from source
1. run `bash tests/scripts/task_unit_test.sh pytorch gpu`


```shell
bash tests/scripts/task_unit_test.sh pytorch gpu
```
```shell
FAILED tests/python/common/test_heterograph.py::test_convert[idtype0] - AttributeError: module 'numpy' has no attribute 'asscalar'
FAILED tests/python/common/test_heterograph.py::test_convert[idtype1] - AttributeError: module 'numpy' has no attribute 'asscalar'
```

## Expected behavior

The test will fail because the numpy has stopped supporting asscalar.

## Environment

 - DGL Version (e.g., 1.0): 1.1
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3): torch: 2.0.0
 - OS (e.g., Linux): Linux
 - How you installed DGL (`conda`, `pip`, source): source
 - Build command you used (if compiling from source): `cmake -DUSE_CUDA=ON ..`
 - Python version: 3.8
 - CUDA/cuDNN version (if applicable): 12.1
 - GPU models and configuration (e.g. V100): NVIDIA GeForce RTX 3060

--- a/tests/python/common/test_heterograph.py
+++ b/tests/python/common/test_heterograph.py
@@ -1313,8 +1313,8 @@ def test_convert(idtype):
         dsttype = hg.ntypes[ntype_id[dst[i]]]
         etype = hg.etypes[etype_id[i]]
         src_i, dst_i = hg.find_edges([eid[i]], (srctype, etype, dsttype))
-        assert np.asscalar(F.asnumpy(src_i)) == nid[src[i]]
-        assert np.asscalar(F.asnumpy(dst_i)) == nid[dst[i]]
+        assert F.asnumpy(src_i).item() == nid[src[i]]
+        assert F.asnumpy(dst_i).item() == nid[dst[i]]
 
     mg = nx.MultiDiGraph(
         [

